Add Index controller documentation (#26)
kuzzleio/sdk-csharp: doc/1/controllers/index/set-auto-refresh/index.md

---
code: true
type: page

> title: setAutoRefresh

Reviewer: ```suggestion
title: SetAutoRefreshAsync
```

description: Set the autorefresh flag
---

> # setAutoRefresh(index, autorefresh, [options])

Reviewer: ```suggestion
# SetAutoRefreshAsync
```

> The setAutoRefresh action allows to set the autorefresh flag for the index.

Reviewer 2: ```suggestion
The setAutoRefresh action updates the autorefresh flag for the provided index.
```

Each index has an autorefresh flag.  

> When set to true, each write request trigger a [refresh](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) action on Elasticsearch.  

Reviewer 2: ```suggestion
When set to true, each write request triggers a [refresh](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) action on Elasticsearch.  
```

> When set to true, each write request triggers a [refresh](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) action on Elasticsearch.  

Reviewer 2: do not use the current ES documentation

> Without a refresh after a write request, the documents may not be immediately visible in search.

Reviewer 2: ```suggestion
Without a refresh after a write request, the documents may not be immediately available in search results.
```

:::info
A refresh operation comes with some performance costs.  
While forcing the autoRefresh can be convenient on a development or test environment,  
we recommend that you avoid using it in production or at least carefully monitor its implications before using it.
:::

## Arguments

```cs
Task SetAutoRefreshAsync(string index , boolean autoRefresh);
```

> | Arguments     | Type                       | Description       |

Reviewer: ```suggestion
| Argument      | Type                       | Description       |
```
